### DISCUSSION TITLE
Code may not be accepted by the compiler in the future

### DISCUSSION THREAD

**Discussion by J0HN50N133** [GreptimeTeam/greptimedb#4407] [General]
https://github.com/GreptimeTeam/greptimedb/blob/2ae2a6674e8a780cb93723de9b749b65ea28562e/src/datatypes/src/macros.rs#L43

What does `$_` exactly means here? It's missing fragment specifier, and [may not be accepted by the compiler in the future](https://github.com/rust-lang/rust/pull/128006).

@killme2008

**Comment by evenyag** [comment]
It is the captured `$` so `$S` should become `$Int8Type`, `$Int16Type` after the outer macro expanded.
https://github.com/GreptimeTeam/greptimedb/blob/547730a467fe14ba3390f1f1405c5731f445c8aa/src/common/function/src/scalars/aggregate/argmax.rs#L119-L132

If I understood correctly......
